clippy_lints/src/copies.rs

```diff
@@ -365,11 +375,18 @@ where
     None
 }
 
-fn search_same_list<T, Hash, Eq>(exprs: &[T], hash: Hash, eq: Eq) -> Option<Vec<(&T, &T)>>
+fn search_same_list<T, Hash, Eq>(exprs: &[T], hash: Hash, eq: Eq) -> Vec<(&T, &T)>
 where
     Hash: Fn(&T) -> u64,
     Eq: Fn(&T, &T) -> bool,
 {
+
+    let matched_expr = search_common_cases(&exprs, &eq);
```

Comment:
You could use `if let` to circumvent the use of `unwrap`.

```suggestion
    if let Some(expr) = search_common_cases(&exprs, &eq) {
        return vec![expr];
    }
```